### DISCUSSION TITLE
Setup: Raspi 4 + LibreELEC + APA102

### DISCUSSION THREAD

**Discussion by PaulePils** [awawa-dev/HyperHDR#798] [Q&A] [answered]
I recently switched from Hyperion to HyperHDR. Till now, all works but i think I have the feeling that there is still room for improvement.

1. The strip is directly connected to the Physical GPIO Pin 6, 19, 23. Is this still correct for HyperHDR?
2. Would a ESP32 be better?
3. I don't have a level shifter because Hyperion never mentioned it when I started the project... How can I connect it? Could I solder it in between the cables? A prototyp board seams a bit large for just one shifter...
4. I installed the addon to libreELEC. Could I update HyperHDR via the addon or do I need to copy the files everytime?

**Comment by awawa-dev** [comment, marked as answer]
Hi
If you managed to turn on the LED without a level shifter, it's probably OK and you don't need to do anything else. Theoretically, it may happen that the LED strip will not recognize frames from time to time due to too low voltage, but apart from observation (frozen LED strip) there is no other way to diagnose it. After v20 release there will be also updated v20 plugin for LibreELEC.

**Reply by PaulePils** [reply]
Thanks for the answer. Frozen LED have not occured until not but I often have these errors in the log
> [V4L2:USB3 VIDEO: US] Frame too small: 0 != 960000

To use the LUT calibration. I only need to hook the raspi with scraper to an HDR monitor or is only the scraper to my PC enough?

**Reply by awawa-dev** [reply]
It should be the same display you use for LibreELEC

**Reply by PaulePils** [reply]
But the calculation is done by the libreELEC or the machine that shows the webpage?
Sorry for asking this trivial question

**Reply by awawa-dev** [reply]
The device that runs HyperHDR

**Reply by PaulePils** [reply]
Okay thanks :-)
Then I will try it out during the night
